client/azure.go

```diff
@@ -70,6 +70,8 @@ type AzureClientSet struct {
 	DNSZonesClient *dns.ZonesClient
 	// InterfacesClient manages virtual network interfaces.
 	InterfacesClient *network.InterfacesClient
+	// VirtualNetworkClient manages virtual networks.
+	VirtualNetworkClient *network.VirtualNetworksClient
```

Comment:
To make the `vnetpeeringv2` resource resilient against timing issues due to the non blocking resource implementations we have to check for virtual networks. Thus the need for the client.